src/chain/network_event.rs

```diff
@@ -59,10 +57,24 @@ pub struct SendAckMessagePayload {
 
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord, Hash, Clone)]
 pub struct SectionInfoSigPayload {
+    /// The public key share for that signature share
+    pub pub_key_share: BlsPublicKeyShare,
     /// The signature share signing the SectionInfo.
-    pub share: (BlsPublicKeyShare, BlsSignatureShare),
-    /// The key set to combine shares.
-    pub pk_set: BlsPublicKeySet,
+    pub sig_share: BlsSignatureShare,
+}
+
+impl SectionInfoSigPayload {
+    pub fn new(
+        info: &SectionInfo,
+        full_id: &FullId,
+    ) -> Result<SectionInfoSigPayload, RoutingError> {
+        let proof = Proof::new(*full_id.public_id(), full_id.signing_private_key(), &info)?;
+
+        Ok(SectionInfoSigPayload {
+            pub_key_share: BlsPublicKeyShare(proof.pub_id),
```

Comment:
just a curious question, if we only have the singing pub_id & sig included in this struct, why still call it `share`?

Reply:
because it will be a BLS signature share and public key share (i.e a part of the sig needed to identify it as signed by the section).